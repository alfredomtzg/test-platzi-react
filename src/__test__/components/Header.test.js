import React from 'react';
import { mount, shallow } from 'enzyme';
import ProviderMock from '../../__mocks__/ProviderMock';
import { create } from 'react-test-renderer';
import Header from '../../components/Header';

describe('<Header/>', () => {
  test('should Render Header', () => {
    const header = shallow(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.length).toEqual(1);
  });
  test('should have title ', () => {
    const header = mount(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.find(".Header-title").text()).toEqual("Platzi Store")
  });
});

// Snapshot the header

describe('Snapshot of header', () => {
  test('should be not change the style in header', () => {
    const header = create(
      <ProviderMock>
        <Header />
      </ProviderMock>,
    );
    expect(header.toJSON()).toMatchSnapshot();
  });

});
