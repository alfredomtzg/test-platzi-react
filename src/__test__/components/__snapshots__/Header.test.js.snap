// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of header should be not change the style in header 1`] = `
<div
  className="Header"
>
  <h1
    className="Header-title"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Platzi Store
    </a>
  </h1>
  <div
    className="Header-checkout"
  >
    <a
      href="/checkout"
      onClick={[Function]}
    >
      <i
        className="fas fa-shopping-basket"
      />
    </a>
  </div>
</div>
`;
